lib/pleaserun/detector.rb

```diff
@@ -25,14 +25,19 @@ def detect_platform
 
   def detect_systemd
     # Expect a certain directory
-    return false unless File.directory?("/lib/systemd/system")
+    return false unless File.directory?("/lib/systemd/system") || File.directory?("/usr/lib/systemd/system")
+
+    # Verify presence of systemctl before proceeding.  This catches an edge
+    # case in ubuntu 12.04 where the systemctl path exists, but no binary
+    out, status = execute([ "which", "systemctl" ])
```

Comment:
Not all systems have `which` available. Thoughts on having `execute` check for this - I'm thinking scanning ENV["PATH"] manually for the program and returning nonzero status if not found.